crates/turbo-tasks-memory/src/task.rs

```diff
@@ -1182,15 +1166,25 @@ impl Task {
     }
 
     /// Access to a cell.
-    pub(crate) fn with_cell_mut<T>(&self, index: usize, func: impl FnOnce(&mut Cell) -> T) -> T {
+    pub(crate) fn with_cell_mut<T>(&self, index: CellId, func: impl FnOnce(&mut Cell) -> T) -> T {
         let mut state = self.state.write();
-        func(&mut state.created_cells[index])
+        let list = state.cells.entry(index.type_id).or_default();
+        let i = index.index as usize;
+        if list.len() <= i {
+            list.resize_with(i + 1, Default::default);
```

Comment:
Can you leave a comment why this would happen? Seems like this could cause a lost of wasted space.

Reply:
It shouldn't happen in practive since index comes from a counter which is only increasing by one each time.